Source/Control/adsr.h

```diff
@@ -11,18 +11,14 @@ namespace daisysp
 - IDLE   = located at phase location 0, and not currently running
 - ATTACK  = First segment of envelope where phase moves from 0 to 1
 - DECAY   = Second segment of envelope where phase moves from 1 to SUSTAIN value
-- SUSTAIN = Third segment of envelope, stays at SUSTAIN level until GATE is released
 - RELEASE =     Fourth segment of envelop where phase moves from SUSTAIN to 0
-- LAST    =  Last segment, aka release
 */
 enum
 {
-    ADSR_SEG_IDLE,
-    ADSR_SEG_ATTACK,
-    ADSR_SEG_DECAY,
-    ADSR_SEG_SUSTAIN,
-    ADSR_SEG_RELEASE,
-    ADSR_SEG_LAST,
+    ADSR_SEG_IDLE    = 0,
+    ADSR_SEG_ATTACK  = 1,
```

Comment:
Can we remove the `= x`? Now that the enum is not used for array adressing anymore, the numbers have become irrelevant.

Reply:
I've added this to keep the interface of this class stable. The method GetCurrentSegment() returns an uint8_t and it's not unlikely, that somebody compares this value against a magic number. I know, it's not pretty but this way it doesn't break other projects.